packages/phone-number-privacy/combiner/src/common/crypto-clients/domain-crypto-client.ts

```
@@ -0,0 +1,39 @@
+import { ErrorMessage, KeyVersionInfo, PoprfCombiner } from '@celo/phone-number-privacy-common'
+import Logger from 'bunyan'
+import { CryptoClient } from './crypto-client'
+
+export class DomainCryptoClient extends CryptoClient {
+  private poprfCombiner: PoprfCombiner
+
+  constructor(protected readonly keyVersionInfo: KeyVersionInfo) {
+    super(keyVersionInfo)
+    this.poprfCombiner = new PoprfCombiner(keyVersionInfo.threshold)
+  }
+
+  protected get allSignaturesLength(): number {
+    // No way of verifying signatures on the server-side
+    return this.unverifiedSignatures.length
+  }
+
+  private get allSigsAsArray(): Uint8Array[] {
+    return this.unverifiedSignatures.map((response) => Buffer.from(response.signature, 'base64'))
+  }
+
+  /*
+   * Aggregates blind partial signatures into a blind aggregated POPRF evaluation.
+   * On error, logs and throws exception for not enough signatures.
+   * Verification of partial signatures is not possible server-side
+   * (i.e. without the client's blinding factor).
+   */
+  protected _combinePartialBlindedSignatures(_blindedMessage: string, logger: Logger): string {
+    try {
+      const result = this.poprfCombiner.blindAggregate(this.allSigsAsArray)
+      if (result !== undefined) {
+        return result.toString('base64')
+      }
+    } catch (error) {
+      logger.error(error)
```

Eating the error here and continuing with a throw of `NOT_ENOUGH_PARTIAL_SIGNATURES` is a bit funky. In particular, it would lead to issues if the array of signatures contains an invalid point (i.e. the array member cannot be decoded to a valid group element). If this is the case, ever call to this function with an array containing the bad element will throw `NOT_ENOUGH_PARTIAL_SIGNATURES`, which is misleading.

---

Totally -- what do you think about this (in an effort to balance getting this merged but also tackling the low-hanging fruit around error stuff):
- we propagate the actual errors in the crypto client combine functions (address this pre-merge)
- ensure that we are at least logging errors in places where there are general catches (i.e. in `requestHasValidKeyVersion` and `responseHasExpectedKeyVersion`) (address this pre-merge)
- create a ticket to revisit & improve error handling more generally in ODIS (i.e. redoing the handleMissingSigs & errorCodeToError logic to be more useful) (address this *post*-merge)